src/Base/Repositories/BaseRepository.php

```diff
@@ -38,9 +38,5 @@ public function update(int $id, array $attributes): bool
         return $model->update($attributes);
     }
 
-    public function delete(int $id): bool
-    {
-        $model = $this->model()->find($id);
-        return $model->delete();
-    }
+    abstract public function delete(int $id): bool;
```

Comment:
What's the benefit of making this method abstract, vs. having a base method that would be sufficient for most implementations?

Reply:
Read the body on the commit for this change

Reply:
Good point!

If and when more than one repository uses the exact same basic deletion logic, we should, in theory, be able to remove the `abstract` keyword and implement the base method without making any other changes, right?

Reply:
Correct